crates/ra_mbe/src/mbe_expander/transcriber.rs

```diff
@@ -51,15 +54,21 @@ pub(super) fn transcribe(
     bindings: &Bindings,
 ) -> Result<tt::Subtree, ExpandError> {
     assert!(template.delimiter == None);
-    let mut ctx = ExpandCtx { bindings: &bindings, nesting: Vec::new(), var_expanded: false };
+    let mut ctx = ExpandCtx { bindings: &bindings, nesting: Vec::new() };
     expand_subtree(&mut ctx, template)
 }
 
+#[derive(Debug)]
+struct NestingState {
+    idx: usize,
+    hit: bool,
```

Comment:
OTOH, would you mind add some comment about `s.hit` ?

Reply:
`hit` is also a marker like `at_end`, it's present to mark the use of a given nested level (i.e. prevent an infinite loop if there is no variable in a repetition).